sdk/src/main/scala/com/horizen/storage/leveldb/VersionedLevelDbStorageAdapter.scala

```diff
@@ -48,25 +54,41 @@ class VersionedLevelDbStorageAdapter(pathToDB: String, keepVersions: Int) extend
     //key for storing version shall not be used as key in any key-value pair in VersionedLDBKVStore
     require(!toUpdateAsScala.exists(pair => pair.getKey == version) && !toRemoveAsScala.contains(version))
 
+    //@TODO added for compatibility with LSMStore, probably interface shall be changed to work with map collection for toUpdate/toremove
+    require(toUpdateAsScala.map(_.getKey).toSet.size == toUpdateAsScala.size, "duplicate key in `toUpdate`")
+    require(toRemoveAsScala.toSet.size == toRemoveAsScala.size, "duplicate key in `toRemove`")
+
+    require(!versionIsExistInStorage(version), "Version is already exist in storage")
+
     val convertedToUpdate = toUpdateAsScala.map(pair => (pair.getKey.data, pair.getValue.data))
     val convertedToRemove = toRemoveAsScala.map(_.data)
     dataBase.update(convertedToUpdate, convertedToRemove)(version)
   }
 
-  override def rollback(versionID: ByteArrayWrapper): Unit = dataBase.rollbackTo(versionID)
+  def versionIsExistInStorage(versionForSearch: ByteArrayWrapper): Boolean = {
```

Comment:
maybe make it `private` and rename to `isVersionExistInStorage` or even `isVersionExist`

Reply:
done